torch/nn/parameter.py

```diff
@@ -3,7 +3,18 @@
 from collections import OrderedDict
 
 
-class Parameter(torch.Tensor):
+class ParameterMeta(type):
+    # Make `isinstance(t, Parameter)` return True for custom tensor instances that have the _is_param flag.
+    def __instancecheck__(self, instance):
+        return super().__instancecheck__(instance) or (hasattr(instance, '_is_param') and instance._is_param)
+
+
+# Metaclass to combine _TensorMeta and the instance check override for Parameter.
+class TensorParameterMeta(torch._C._TensorMeta, ParameterMeta):
+    pass
```

Comment:
Curious how come ParameterMeta can't directly inherit from TensorMeta

Reply:
Turns out it can! :)